### ISSUE TITLE
[CAL-1364] availability override not working

### ISSUE THREAD

**Issue by Jaibles** [calcom/cal.com#8000]
Customer reported that

> "I just had an issue where the override did not work... someone was able to reschedule a call to a time that should not have been available

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-1364](https://linear.app/calcom/issue/CAL-1364/availability-override-not-working)</sub>

**Comment by ilyasubkhankulov** [comment]
I'd like to confirm this report: availability overrides do not respect timezones on the booking page.

To reproduce:
1. set a date override for a day and hit "save"
2. go to the booking page and change the timezone dropdown
3. the available booking times do not adjust for the user's timezone.
4. an attempt to book during an incorrectly available time slot yields an error "Could not book the meeting. Can you try again with a different time? Go back."

**Comment by ilyasubkhankulov** [comment]
I believe the issue/bug is somewhere in [this part](https://github.com/calcom/cal.com/blob/ae5425f908fe4b6cef2c7ca4c7d07938a89fb7be/packages/core/getUserAvailability.ts#L216) of getUserAvailability.ts

---
My troubleshooting notes:

Testing availability slots with no date override for a particular day:
1. availability set by user in PST: 11:00 - 13:00 (UTC: 18:00-20:00)
2. availability in Baku from API: 18:00-20:00
3. availability in Baku from UI: 22:00-00:00 AZT GMT+4 (UTC 18:00-20:00) 

Testing availability slots with a date override set in the DB as 13:00:00 - 14:00:00 PT for a particular day:
1. availability set by user in PST: 13:00 - 14:00 (UTC: 20:00-21:00)
2. availability in Baku from API: 9:00-10:00 **<--- INCORRECT?** 
3. availability in Baku from UI: 13:00-14:00 AZT GMT+4 (UTC 9:00-10:00)

**Comment by Rphad** [comment]
I got a similar problem. When trying to book on one of my overridden dates, the following error happens : "Could not book the meeting. Can you try again with a different time? Go back"

![Screenshot of the error](https://user-images.githubusercontent.com/49572419/230283480-2ec9db0c-568b-4923-8586-e2f8d9be5936.jpg)

**Comment by Karanjot-singh** [comment]
@Jaibles 
A related issue that I noticed:
If I override my availability on a date. The time slots on those dates stay the same even if the person booking the time changes their timezone. i.e. it always displays overridden time in my timezone only (IST).
 
On trying to book that time an error is thrown:

Could not book the meeting. Can you try again with a different time? Go back.

**Comment by jacobzukerman** [comment]
Adding another issue to this thread, but can create a separate bug if that is helpful. We just ran into this use via one of our Cal users (we have a self-hosted experience):

The user created an override that extended the time beyond the default availability:
- Default availability for Thursday: 2 - 4 PM
- Override for one specific Thursday: 1:30 - 4 PM

Cal showed 1:30 PM free, but when booking, threw a "Could not book..." error. When I extended the default availability to 1:30 PM, it worked fine. Did not like that extended overlap from the default hours set.
